stdlib/EvaluatorSettings.pkl

```diff
@@ -0,0 +1,145 @@
+//===----------------------------------------------------------------------===//
+// Copyright © 2024 Apple Inc. and the Pkl project authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//===----------------------------------------------------------------------===//
+
+/// Common settings for Pkl's own evaluator.
+@ModuleInfo { minPklVersion = "0.26.0" }
+module pkl.EvaluatorSettings
+
+/// The external properties available to Pkl, read using the `prop:` scheme.
+externalProperties: Mapping<String, String>?
+
+/// The environment variables available to Pkl, read using the `env:` scheme.
+///
+/// Example:
+/// ```
+/// env {
+///   ["IS_PROD"] = "true"
+/// }
+/// ```
+env: Mapping<String, String>?
+
+/// The set of module URI patterns that can be imported.
+///
+/// Each element is a regular expression pattern that is tested against a module import.
+///
+/// Modules are imported either through an amends or extends clause, an import clause, or an
+/// import expression.
+///
+/// Example:
+/// ```
+/// allowedModules {
+///   "https:"
+///   "file:"
+///   "package:"
+///   "projectpackage:"
+/// }
+/// ```
+allowedModules: Listing<String(isRegex)>?
+
+/// The set of resource URI patterns that can be imported.
+///
+/// Each element is a regular expression pattern that is tested against a resource read.
+///
+/// Example:
+/// ```
+/// allowedResources {
+///   "https:"
+///   "file:"
+///   "package:"
+///   "projectpackage:"
+///   "env:"
+///   "prop:"
+/// }
+/// ```
+allowedResources: Listing<String(isRegex)>?
+
+/// Disables the file system cache for `package:` modules.
+///
+/// When caching is disabled, packages are loaded over the network and stored in memory.
+noCache: Boolean?
+
+/// A collection of jars, zips, or directories to be placed into the module path.
+///
+/// Module path modules and resources may be read and imported using the `modulepath:` scheme.
+///
+/// Relative paths are resolved against PklProject's enclosing directory.
```

Comment:
This documentation doesn't make that much sense here.

I think let's add these details to the doc comment of property `evaluatorSettings` in `pkl:Project`.

Same goes for the comments for `moduleCacheDir` and `rootDir`.